### ISSUE TITLE
PEM read failures no longer include error messages

### ISSUE THREAD

**Issue by ummmmm** [openssl/openssl#20665]
Commit 7bc027d caused PEM read failures to no longer include an error message.
OpenSSL 1.1.x `PEM_read_bio_PrivateKey` called `PEM_bytes_read_bio_secmem` and if that failed return out without clearing error messages.

OpenSSL 3.x `PEM_read_bio_PrivateKey` has a call chain of `PEM_read_bio_PrivateKey_ex` -> `pem_read_bio_key` -> `pem_read_bio_key_legacy`.

`pem_read_bio_key_legacy` includes calls to `ERR_set_mark` and `ERR_pop_to_mark` when either `PEM_bytes_read_bio_secmem` or `PEM_bytes_read_bio` fails.

Calling `PEM_read_bio_PrivateKey` in OpenSSL 1.x would have available error messages whereas OpenSSL 3.x there are no error messages for the same failure.

**Comment by nhorman** [comment]
this was introduced in commit fbbd425336144455f4a976acd7b890352ef7ed38 as a fix for https://github.com/openssl/openssl/pull/15949

Seems like the suppression of spurious error messages masked legitimate error messages.

Not sure what to do about this, as there is no good way to delineate 'ok' errors from 'real' error.  @t8m ?

**Comment by t8m** [comment]
Yeah, IMO we would have to somehow recognize which errors are fatal (i.e. not related to the particular decoding process failing because of the type mismatch) and keep these.
